### PR TITLE
Code maintenance

### PR DESCRIPTION
**Pull request by jlmaurer**
## Description

- Updates unit tests
- Removes unused functions
- Adds docstrings
- Update documentation as needed

## Motivation and Context

- Increase code coverage to 60%
- Add docstrings for important functions
- Clean up old code

## How Has This Been Tested?

- tests pass locally

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have added an explanation of what your changes do and why you'd like us to include them.
- [x] I have written new tests for your core changes, as applicable.
- [x] I have successfully ran tests with your changes locally.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by jlmaurer** [comment]
Leaving a note that several unit tests using GMAO are failing randomly/

File: tools/RAiDER/dem.py [hunk 1]
```diff
@@ -14,47 +14,17 @@
...
 
 
 def download_dem(
-    ll_bounds,
-    writeDEM=False,
-    outName='warpedDEM',
-    buf=0.02,
-    overwrite=False,
-):
+        ll_bounds=None,
```
Comment (bbuzz31) [inline tools/RAiDER/dem.py:22]: what happens if you don't pass bounds now? will it download global?

**Review by bbuzz31 — APPROVED** [review]
LGTM pending circleci success

**Comment by jhkennedy** [comment]
@jeremy, looking at the failures in CircleCI, which is testing in Python 3.12, I'm seeing this breakdown:

* `test/test_intersect.py FF`: caused by #607 
* `test/test_slant.py FF`: caused by #607 
* `test/test_synthetic.py sssFFFFFFFFF`: caused by #607
* `test/test_temporal_interpolate.py FF`: HTTP 301 Moved Permanently -- the data were trying to access seems to be missing/moved:
  ```
  webob.exc.HTTPError: 301 Moved Permanently
  <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
  <html><head>
  <title>301 Moved Permanently</title>
  </head><body>
  <h1>Moved Permanently</h1>
  <p>The document has moved <a href="https://opendap.nccs.nasa.gov/dods/GEOS-5/fp/0.25_deg/assim/inst3_3d_asm_Nv.dds">here</a>.</p>
  </body></html>
  ```
  and
  ```
  webob.exc.HTTPError: 301 Moved Permanently
  <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
  <html><head>
  <title>301 Moved Permanently</title>
  </head><body>
  <h1>Moved Permanently</h1>
  <p>The document has moved <a href="https://opendap.nccs.nasa.gov/dods/GEOS-5/fp/0.25_deg/assim/inst3_3d_asm_Nv.dds">here</a>.</p>
  </body></html>
  ```

I can likely have a PR open to fix #607 tomorrow.

**Comment by jhkennedy** [comment]
Merging and opening a follow on PR.

**Pull request merged**